### PR TITLE
Epic popups updates for version 3.8.0 (main <- main-beta)

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

File: src/modules/esl-utils/dom/rect.ts [hunk 1]
```diff
@@ -19,6 +19,15 @@ export class Rect {
     return new this(rect.x || rect.left, rect.y || rect.top, rect.width, rect.height);
   }
 
+  /**
+   * The static method checks the equality of the two Rect instances.
+   * @param rect1 - first instance of Rect
+   * @param rect2 - second instance of Rect
+   * */
```
Comment (abarmina) [inline src/modules/esl-utils/dom/rect.ts:26]:
```suggestion
   */
```

**Review by abarmina — APPROVED** [review]

File: pages/views/examples/popup-transition.njk [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      <esl-trigger target="#popup3" mode="toggle" class="btn btn-sec-blue">Left</esl-trigger>
+    </div>
+    <div class="tile">
+      <esl-trigger target="#popup4" mode="toggle" class="btn btn-sec-blue">Right</esl-trigger>
+    </div>
+  </div>
+</section>
+
+<h2>Fast resizing</h2>
+<section class="row"">
```
[the remark this answers is not in the document]
Comment (dshovchko) [reply pages/views/examples/popup-transition.njk:95]: fixed in https://github.com/exadel-inc/esl/pull/711

File: pages/views/examples/popup-transition.njk [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+      <esl-trigger target="#popup3" mode="toggle" class="btn btn-sec-blue">Left</esl-trigger>
+    </div>
+    <div class="tile">
+      <esl-trigger target="#popup4" mode="toggle" class="btn btn-sec-blue">Right</esl-trigger>
+    </div>
+  </div>
+</section>
+
+<h2>Slow vertical shifting</h2>
+<section class="row"">
```
[the remark this answers is not in the document]
Comment (dshovchko) [reply pages/views/examples/popup-transition.njk:131]: fixed in https://github.com/exadel-inc/esl/pull/711

File: pages/views/examples/popup-transition.njk [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      <esl-trigger target="#popup3" mode="toggle" class="btn btn-sec-blue">Left</esl-trigger>
+    </div>
+    <div class="tile">
+      <esl-trigger target="#popup4" mode="toggle" class="btn btn-sec-blue">Right</esl-trigger>
+    </div>
+  </div>
+</section>
+
+<h2>Slow resizing</h2>
+<section class="row"">
```
[the remark this answers is not in the document]
Comment (dshovchko) [reply pages/views/examples/popup-transition.njk:149]: fixed in https://github.com/exadel-inc/esl/pull/711

File: pages/views/examples/popup-transition.njk [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+      <esl-trigger target="#popup3" mode="toggle" class="btn btn-sec-blue">Left</esl-trigger>
+    </div>
+    <div class="tile">
+      <esl-trigger target="#popup4" mode="toggle" class="btn btn-sec-blue">Right</esl-trigger>
+    </div>
+  </div>
+</section>
+
+<h2>Slow horizontal shifting</h2>
+<section class="row"">
```
[the remark this answers is not in the document]
Comment (dshovchko) [reply pages/views/examples/popup-transition.njk:113]: fixed in https://github.com/exadel-inc/esl/pull/711

File: src/modules/esl-utils/dom/rect.ts (hunk 1 above)
Comment (dshovchko) [reply src/modules/esl-utils/dom/rect.ts:26]: fixed in https://github.com/exadel-inc/esl/pull/711

File: src/modules/esl-popup/core/esl-popup.ts [hunk 6]
```diff
@@ -156,65 +196,118 @@ export class ESLPopup extends ESLToggleable {
...
+  /** Actions to execute on activator scroll event. */
   @bind
-  protected onActivatorScroll(e: Event) {
+  protected onActivatorScroll(e: Event): void {
+    if (this._updateLoopID) return;
     this._updatePosition();
   }
 
+  /** Creates listeners and observers to observe activator after showing popup */
   protected _addActivatorObserver(target: HTMLElement) {
```
Comment (yadamskaya) [inline src/modules/esl-popup/core/esl-popup.ts:223]:
```suggestion
  protected _addActivatorObserver(target: HTMLElement): void {
```

**Review by yadamskaya — APPROVED** [review]

**Pull request merged**